### PR TITLE
fix: [PL-32504]: Do not check for userIn Ng when publishing login events in case of enableAudit is false

### PR DESCRIPTION
**Pull request by VikasMaddukuriHarness**
## Describe your changes
This scenario is happening in SMP case :
If a customer is not part of ng and when login happens , we are making a Rest call to ng-manager to get whether the user is part of ng to publish audit. As ng-manager is not up, this Rest call is timing out and this is logging exception. This log is flooding customer logs. This fix will check whether the enable audit is set to true or not and then only make a rest call to Ng.

Testing:
1. Verified whether REst call is being made when enableAudit is set to true and vice-versa
2. Also verified whether the outbox event got consumed and removed from dB or not
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46574)

### CODE REVIEW

File: 400-rest/src/main/java/software/wings/core/outbox/UserEventHandler.java [hunk 1]
```diff
@@ -31,18 +31,24 @@
...
+
 import com.google.inject.Inject;
 import com.google.inject.name.Named;
 import lombok.extern.slf4j.Slf4j;
 
 @Slf4j
 public class UserEventHandler implements OutboxEventHandler {
   private final AuditClientService auditClientService;
-  @Inject @Named("PRIVILEGED") private UserMembershipClient userMembershipClient;
+  @Inject private final MainConfiguration mainConfiguration;
```
Comment (nishantsaini20) [inline 400-rest/src/main/java/software/wings/core/outbox/UserEventHandler.java:43]: When using constructor injection you don't need to use `@Inject` annotation here.

File: 400-rest/src/main/java/software/wings/core/outbox/UserEventHandler.java [hunk 2]
```diff
@@ -31,18 +31,24 @@
...
+  @Inject @Named("PRIVILEGED") private final UserMembershipClient userMembershipClient;
```
Comment (nishantsaini20) [inline 400-rest/src/main/java/software/wings/core/outbox/UserEventHandler.java:44]: Same here. Also `@Named` annotation can be moved to constructor argument.

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]

File: 400-rest/src/test/java/software/wings/core/outbox/UserEventHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+public class UserEventHandlerTest extends WingsBaseTest {
+  @Mock private AuditClientService auditClientService;
+  private UserEventHandler userEventHandler;
+  @Mock private UserMembershipClient userMembershipClient;
+  private MainConfiguration mainConfiguration;
+
+  @Before
+  public void setup() {
+    MockitoAnnotations.initMocks(this);
+    mainConfiguration = mock(MainConfiguration.class);
```
Comment (nishantsaini20) [inline 400-rest/src/test/java/software/wings/core/outbox/UserEventHandlerTest.java:53]: `mock()` is equivalent to `@Mock` annotation. So not required.

File: 400-rest/src/test/java/software/wings/core/outbox/UserEventHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    auditClientService = mock(AuditClientService.class);
+    userMembershipClient = mock(UserMembershipClient.class);
```
Comment (nishantsaini20) [inline 400-rest/src/test/java/software/wings/core/outbox/UserEventHandlerTest.java:55]: same

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]

**Review by nishantsaini20 — DISMISSED** [review]

**Review by pbarapatre10 — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46574)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Pull request merged**